eclair-node/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala

```diff
@@ -33,6 +35,17 @@ class LocalPaymentHandler extends Actor with ActorLogging {
       sender ! h
       context.become(run(h2r + (h -> r)))
 
+    case NewPaymentRequest(amount) if amount.amount > 0 && amount.amount < 4294967295L =>
+      val r = generateR
+      val h: BinaryData = Crypto.sha256(r)
+      val pr = s"${nodeParams.privateKey.publicKey}:${amount.amount}:${h.toString}"
```

Comment:
Please isolate this in a static method of `object LocalPaymentHandler` for better testability.